client/src/utils/contexts/MapContext.js

```diff
@@ -1,10 +1,25 @@
+class MapSearchInstance{
+    constructor(locationName,lat,long){
+      this.locationName = locationName;
+      this.lat = lat;
+      this.long = long;
+    }
+}
+// FIX LATE ADDRESS DISPLAY IN DOCTOR REGISTER PAGE
 export async function searchToAddressResults(searchAddress) {
     try {
-        const getAddressResult = await fetch(
-          `https://geocoder.ls.hereapi.com/search/6.2/geocode.json?languages=en-US&maxresults=4&searchtext=${searchAddress}&apiKey=${process.env.REACT_APP_HERE_API_KEY}`
-        );
-        return getAddressResult;
+        searchAddress = searchAddress.replaceAll(' ', '+');
```

Comment:
Try finding a react/js function which encodes a string url safe. This depending on how handled at api might wrong results for special characters like comma, dot, colon etc. If it is working fine for special characters this is good.